src/dfi/validation.cpp

```diff
@@ -4666,6 +4685,15 @@ Res ProcessDeFiEventFallible(const CBlock &block,
     // Construct undo
     FlushCacheCreateUndo(pindex, mnview, cache, uint256S(std::string(64, '1')));
 
+    // Ocean archive
```

Comment:
Maybe this should be after evm_try_unsafe_commit_block in ConnectBlock, that's the very last point of failure when connecting new blocks.